src/plugins/discourse/weights.js

```diff
@@ -63,11 +65,26 @@ export const weightsConfigParser: C.Parser<WeightsConfig> = C.fmap(
   upgrade
 );
 
-export function likeWeight(user: ?User): NodeWeight {
+export function likeWeight(
+  {weights = {}}: DiscourseConfig,
+  user: ?User,
+  category: ?DiscourseCategoryId,
+  tags: ?$ReadOnlyArray<Tag>
+): NodeWeight {
   if (user == null) {
     return 0;
   }
-  return _trustLevelWeight(user);
+  if (category == null) {
+    category = -Infinity;
+  }
```

Comment:
Shouldn't this be a type error? CategoryId is a string not a number. Why is Flow OK with this?